service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractRequestHandler.java

```diff
@@ -119,14 +126,32 @@ private void createGetGpcStructuredTask(EhrExtractStatus ehrExtractStatus) {
             getRequiredValue(payload, NHS_NUMBER_PATH),
             getRequiredValue(header, FROM_PARTY_ID_PATH),
             getRequiredValue(header, TO_PARTY_ID_PATH),
-            getRequiredValue(payload, FROM_ASID_PATH),
-            getRequiredValue(payload, TO_ASID_PATH),
+            getFromAsid(getRequiredValue(payload, FROM_ASID_PATH)),
```

Comment:
I don't think we can apply the override here. This will also affect the ASIDs we send back to the requesting system, which if overridden, will be wrong and cause the message to be rejected. We should apply the override when we populate the Ssp headers for the GPC requests.